climpred/prediction.py

```diff
@@ -101,14 +162,37 @@ def compute_hindcast(
 
     Returns:
         skill (xarray object):
-            Predictability with main dimension ``lag``
+            Predictability with main dimension ``lag`` without dimension ``dim``
 
     """
+    if metric in PROBABILISTIC_METRICS:
+        if metric == 'e2r':
+            raise ValueError(
+                'Probabilistic metric `', metric, '` requires comparison `m2r`'
+            )
+        else:
+            stack = False
+    elif dim == 'init':
+        stack = True
+    elif dim == 'member':
```

Comment:
Does there need to be an `else` if dim is neither `init` nor `member`?

Reply:
added